### DISCUSSION TITLE
Bootstrap Version and Vuetify

### DISCUSSION THREAD

**Discussion by Parthian** [stephenmcd/mezzanine#2040] [General]
Hi, 
Installed Mezzanine 6. Did a udemy course (sadly incomplete) but useful as far as it went - the advanced bit never happened.
All seems good. 

I note that Bootstrap is at version v3.2.0 I think. I haven't used bootstrap for a few years. I think v4 or v5 was a complete rewrite.
Any plans to upgrade it?

I have an alternative plan to try and use Vue/Vuetify with Mezzanine. I suspect bootstrap classes will be too integrated but maybe I could just modify the templates as I need them. Building a relatively simple site. Any views?

Stuart

**Comment by henri-hulski** [comment]
I don't think, we gonna update bootstrap as this is a major rewrite.
I mean if someone is ready to do it we should consider it.
To use Mezzanine with React or Vue you should get rid of jquery.

Edit: At least to my knowledge. Maybe it's somehow possible to use them together.

**Reply by Parthian** [reply]
Thanks Henri, 
I was inspired by this course https://www.udemy.com/course/create-a-blog-with-mezzanine-cms-the-best-django-cms/ by Travis Rodgers.
His approach may be common knowledge but involved roughly the following.

1)  python manage.py collecttemplates and then rename the created folder as tempdrafts. This pulls in templates from many Mezzannine folders into one place.
2) In a new App he calls theme create a templates folder. And then when you want something from tempdrafts to override you pull it in with the same folder structure. So if you want to tweak the primary.html you put it in theme/templates/pages/menu
Like so - [https://github.com/rodgtr1/mezzanine-tutorial/tree/Lesson-6/theme/templates](https://github.com/rodgtr1/mezzanine-tutorial/tree/Lesson-6/theme/templates)

But he updated a Bootstrap4Theme to show a new design. Achieved by copying in the bootstrap and jquery static files from the Bootstrap4Theme into the theme/static thus [https://github.com/rodgtr1/mezzanine-tutorial/tree/Lesson-6/theme/static](https://github.com/rodgtr1/mezzanine-tutorial/tree/Lesson-6/theme/static) hunt around in there and you'll find the bootstrap is v4.

Then batter away at base.html to include the new Bootstrap4Theme js and css. Thereafter the various mezzanine templates were tweaked to handle the Bootstrap3 to 4 changes. Which appeared to be fairly minimal. Not suprisingly they will share most of the classes.

That may be obvious stuff. But it indicates to me that it may be reasonable to only change the templates that are needed. Staying in Bootstrap and moving to 5 may be doable with this approach. Probably going to vue/vuetify would be a challenge.

I'm worried about using Mezzanine with jquery which is already all but dead. That may be trickier to remove. It may be that Mezzanine Admin could stay Bootstrap3 and jquery but frontend pages could use more modern frameworks.

**Reply by henri-hulski** [reply]
>  It may be that Mezzanine Admin could stay Bootstrap3 and jquery but frontend pages could use more modern frameworks.

I think this is the way to go. You can actual use a different versions of bootstrap than mezzanine as mezzanine uses a specific namespace for its bootstrap installation or you can use an entirely different framework.
